### ISSUE TITLE
Content hooks are not working with content coming from github repos

### ISSUE THREAD

**Issue by BernardoSM** [nuxt/content#2149]
### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      modules, content
- Runtime Modules:  @nuxt/content@2.7.0
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-kwfyia?file=server%2Fplugins%2Fcontent.ts

### Describe the bug

I was trying to follow this suggestion here https://github.com/nuxt/content/issues/2056#issuecomment-1560200992 and I realized that content hooks wasn't working with me. 

![image](https://github.com/nuxt/content/assets/36281298/8cf6b5fd-4d84-4dee-80ad-da85ed94cede)

So I realized all my content files are fetched by github source like the documentation here https://content.nuxtjs.org/api/configuration#sources so I think there's some relationship.

**Comment by Barbapapazes** [comment]
Hello @BernardoSM ,

_TL;DR: hooks are working fine_

I fork your reproduction and update the code, you can find it [on stckblitz](https://stackblitz.com/edit/github-kwfyia-9raw7g?file=server%2Fplugins%2Fcontent.ts).

You think it was not working because `nuxt/content` is async when it parses content files. So, while you didn't request any file, they are not parsed. For your example, you need to start typing in your search bar in order to do a first query to start parsing.

Once files are parsed, they are cached in `.nuxt/content-cache` using `unjs/unstorage`. If you want to re-parse your files, you can simple delete the folder `.nuxt/content-cache` and fire a new request. `nuxt/content` will detect that files are not parsed and will parsed them again (without reloading the server).

**Comment by BernardoSM** [comment]
Thanks @Barbapapazes it worked! I'll close this issue :D

**Comment by Barbapapazes** [comment]
> Thanks @Barbapapazes it worked! I'll close this issue :D

Nice to hear that! 🙌
